### PR TITLE
Fix conjure type pickling

### LINKED ISSUE #378: Conjure python types are not picklable
## What happened?
#317 broke pickling of conjure types

```
>>> from ontology_metadata_api.ontology_metadata_api import RuleSetBinding
>>> binding = RuleSetBinding(bindings={}, rule_set_rid="some.rid")
>>>
>>> import pickle
>>> pickle.dumps(binding, pickle.HIGHEST_PROTOCOL)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
_pickle.PicklingError: Can't pickle <class 'ontology_metadata_api.ontology_metadata_api.ontology_metadata_api_RuleSetBinding'>: attribute lookup ontology_metadata_api_RuleSetBinding on ontology_metadata_api.ontology_metadata_api failed
```

This is because:

```
>>> binding.__class__
<class 'ontology_metadata_api.ontology_metadata_api.ontology_metadata_api_RuleSetBinding'>
```

`__class__` points to a reference that is not valid

Fix:

```
>>> binding.__class__.__name__
'RuleSetBinding'
>>> binding.__class__.__qualname__
'ontology_metadata_api_RuleSetBinding'
>>> binding.__class__.__qualname__ = binding.__class__.__name__
>>> binding.__class__
<class 'ontology_metadata_api.ontology_metadata_api.RuleSetBinding'>
>>> pickle.dumps(binding, pickle.HIGHEST_PROTOCOL)
b'\x80\x04\x95v\x00\x00\x00\x00\x00\x00\x00\x8c+ontology_metadata_api.ontology_metadata_api\x94\x8c\x0eRuleSetBinding\x94\x93\x94)\x81\x94N}\x94(\x8c\r_rule_set_rid\x94\x8c\x08some.rid\x94\x8c\t_bindings\x94}\x94u\x86\x94b.'
```

### PR DESCRIPTION
**Pull request by JacekLach**
Fixes #378 

## After this PR

==COMMIT_MSG==
Fix pickle serialization failing on conjure generated types.
==COMMIT_MSG==

### CODE REVIEW

File: conjure-python-core/src/test/resources/types/expected/package_name/_impl.py [hunk 1]
```diff
@@ -1214,7 +1238,8 @@ def _options(self, options):
 
 
 product_OptionsUnionVisitor.__name__ = "OptionsUnionVisitor"
-product_OptionsUnionVisitor.__module__ = "OptionsUnionVisitor"
+product_OptionsUnionVisitor.__qualname__ = "OptionsUnionVisitor"
+product_OptionsUnionVisitor.__module__ = "package_name.product"
```
Comment (JacekLach) [inline conjure-python-core/src/test/resources/types/expected/package_name/_impl.py:1242]: drive-by fix

**Review by ahggns — APPROVED** [review]
